Add Citra RNG Setup Guide (#125)
zaksabeast/PokemonRNGGuides: guides/Citra and 3DS/CitraRNG Setup.md

4. Press `A` on the game you want to dump.

5. Select `Manage Title...`.

6. Select `Build CIA (standard)`.

### To dump a game from a cart do the following.

1. Select `[C:] GAMECART`.

2. Select the `trim.3ds` file.

3. Select `NCSD image options...`.

4. Select `Build CIA from file`.

Exit GodMode9 and transfer the files from sdmc:\gm9\out to your PC.

### Setting up Citra

1. Open Citra, under the `File` option choose `Install cia...`.

   - Install both the game(s) and update(s) .cia

2. Place the nand folder from the `sdmc:gm9\out\Citra\user\nand` into your user directory for Citra.
   - For Windows, the path is `C:\Users\[your-user-name]\AppData\Roaming\Citra\nand`
     - The folder AppData is hidden by default, so you need to change the configuration to view it
   - For macOS and Linux, the path is `~/.local/share/citra-emu/nand`
     - The folder .local is hidden on most machines, so you need to change the configuration to view it

### Setting up CitraRNG

Python 3.9 needs to be installed for CitraRNG to work. Python also needs to be added to your `PATH` or else the command will not be recognized. To add Python to your `PATH` make sure to check the box in the installer.

> 1. Open a command prompt anywhere and type `pip install pyside2`.

Reviewer: Only thing i'll be picky on, since last versions use pyside6

---
title: 'CitraRNG Setup'
description: 'Setup Citra for RNG'
slug: 'citrarng-setup'
subCategory: 'Emulator'
---

This guide is for setting up CitraRNG on a PC. An updated CFW (Custom Firmware) console is required to be able to dump system specific information and game data. The console also needs to have the latest game update installed to be dumped.

## Tools

- [Citra](https://citra-emu.org/download/)
- [CitraRNG](https://github.com/Admiral-Fish/CitraRNG)
  - Clone or download the entire repository
- [Python 3.9](https://www.python.org/downloads/)

>   - In the install selection add python to your `PATH`

Reviewer 2: I'm not sure most people are going to know what `PATH` is referring to.  Doesn't this happen automatically with windows?  Or is this referring to the installer?

- [Latest Godmode9](https://github.com/d0k3/GodMode9/releases)
- [Dump_PKMN_Update.gm9 Script](https://cdn.discordapp.com/attachments/389206049401470976/435566091457134598/Dump_PKMN_Updates.gm9)
- A 3DS with CFW (Custom Firmware)
  - https://3ds.hacks.guide/ has instructions for installing CFW
- Latest game update installed on the 3DS

> ## Dumping system and game data

Reviewer 2: I think citra-emu.org has guides for this.  Should we link to them instead of writing out our own instructions?

Part of our Tools includes "Citra", which implies someone already has Citra installed and set up - listing Citra's official guides as bullets to that might be good.

1. Place the `Dump_PKMN_Update.gm9` script in your 3DS sd card in sdmc:\gm9\scripts.
2. Boot your 3DS into GodMode9 by holding `Start` while powering on the 3DS.
3. Press the `Home button` then select `scripts...`.
   - Select `GM9Megascript`
   - Select `Dump Options`
   - Select `Dump Citra Files` to dump Citra files

> 4. Exit back to Godemode9 main menu and select `scripts...` once more.

Reviewer 2: ```suggestion
4. Exit back to the GodMode9 main menu and select `scripts...` once more.
```

>    - Select `Dump_PKMN_Updates` script and follow the on screen directions

Reviewer 2: This looks like a sequential instruction:
4. Go to GodMode9 scripts
5. Select the `Dump_PKMN_Updates` script
6. Dump the game

Should this line be part of the numbered sequence?

5. To dump a game installed on the 3DS do the following.
   - Hover over [A:] SYSNAND SD
   - Hold `R` and press `A` at the same time to open the drive options
   - Select `Open title manager...`
   - Press `A` on the game you want to dump
   - Select `Manage Title...`

>    - Select `Build CIA (standard)`

Reviewer 2: This first line sounds more like a header than part of the sequential list of instructions, and it reads a bit odd to have a numbered header, but the actual instructions are in a bulleted list.

The title of this section is "Dumping system and game data".  What are thoughts on splitting it up into "Dumping system data" and "Dumping game data", then having the bullets of steps 6 and 7 be part of "Dumping game data"?

6. To dump a game from a cart do the following.
   - Select `[C:] GAMECART`
   - Select the `trim.3ds` file
   - Select `NCSD image options...`
   - Select `Build CIA from file`
7. Exit godmode9 and transfer the files from sdmc:\gm9\out to your PC.

## Settings

### Setting up Citra

1. Open Citra, under the `File` option choose `Install cia...`.
   - Install both the game(s) and update(s) .cia
2. Place the nand folder from the `sdmc:gm9\out\Citra\user\nand` into your user directory for Citra.
   - For Windows, the path is `C:\Users\[your-user-name]\AppData\Roaming\Citra\nand`
     - The folder AppData is hidden by default, so you need to change the configuration to view it
   - For macOS and Linux, the path is `~/.local/share/citra-emu/nand`
     - The folder .local is hidden on most machines, so you need to change the configuration to view it

### Setting up CitraRNG

> Python 3.9 needs to be installed for CitraRNG to work. Python also needs to be added to your `Path` or else the command will not be recognized. If it's not installed then it can be added by editing the installation.

Reviewer 2: Whoops, looks like PATH was accidentally not capitalized
```suggestion
Python 3.9 needs to be installed for CitraRNG to work. Python also needs to be added to your `PATH` or else the command will not be recognized. If it's not installed then it can be added by editing the installation.
```

I'm wondering if we should reword this a bit.  At the moment, it sounds like if Python is not installed, then it can be added by editing the installation, but what installation is there to edit if it's not installed?

1. Open a command prompt anywhere and type `pip install pyside2`.
   - This may need administration privileges
2. Copy all the .py files in the CitraRNG folder into `<your Citra directory>/scripting`.

> 3. Open Citra and your Gen 7 Pokémon game, and load your save file.

Reviewer 2: ```suggestion
3. Open Citra and your Gen 7 Pokémon game, then load your save file.
```

   - [This](https://citra-emu.org/wiki/dumping-save-data-from-a-3ds-console/) has instructions for dumping and loading your save file from the 3DS
4. Double-click the citrarng.py file to run the script.

```
Note: If this does not work then right click "citrarng.py" and select "Edit with IDLE". Make sure that you open with Python 3.9 if you have both Python2 and Python3 installed. Then hit "F5" to run the script.

Alternatively, you can run the script by opening a command prompt in the scripting folder and using the command "py citrarng.py".
```

5. In the CitraRNG window select your game and click connect.
   - CitraRNG will update with information from the game such as initial seed, frames, ivs etc.
   - The tool has an auto refresher to allow frames to be updated periodically, this may make the program crash if left on long enough

```

> Note: The higher the delay for the refresh rate the less often the program will crash, 2000 is max. Pause and reconnect CitraRNG every once in a while, to keep it from crashing.

Reviewer 2: ```suggestion
Note: The higher the delay for the refresh rate the less often the program will crash, 2000 is the max. Pause and reconnect CitraRNG every once in a while, to keep it from crashing.
```
